tests/release_tests/test_launch/test_kubernetes.py

```diff
@@ -0,0 +1,162 @@
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-secret-access-key",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                    {
+                                        "name": "AWS_SESSION_TOKEN",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-session-token",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                ]
+                            }
+                        ]
+                    }
+                }
+            },
+        }
+    },
+}
+
+
+@pytest.mark.timeout(180)
+def test_kubernetes_agent_on_local_process(api_key, base_url):
+    if not api_key:
+        api_key = get_wandb_api_key(base_url)
+    aws_id = os.getenv("AWS_ACCESS_KEY_ID")
+    aws_secret = os.getenv("AWS_SECRET_ACCESS_KEY")
+    aws_token = os.getenv("AWS_SESSION_TOKEN")
+
+    create_wandb_and_aws_secrets(NAMESPACE, api_key, aws_id, aws_secret, aws_token)
+
+    try:
+        # Start launch agent
+        agent_process = run_cmd_async(
+            f"wandb launch-agent -q {QUEUE} -e {ENTITY} -c tests/release_tests/test_launch/local-agent-config.yml"
+        )
+
+        # Start run
+        queued_run = launch_add(
+            job=f"{ENTITY}/{PROJECT}/{JOB_NAME}",
+            queue_name=QUEUE,
+            entity=ENTITY,
+            config=LAUNCH_JOB_CONFIG,
+        )
+
+        status, completed_run = wait_for_image_job_completion(
+            NAMESPACE, ENTITY, PROJECT, queued_run
+        )
+
+        summary = completed_run.summary
+        history = completed_run.history(pandas=False)
+
+        assert (
+            status == "Succeeded"
+        ), "Kubernetes job didn't succeed. Check Kubernetes pods and Docker container output."
+        assert summary["time_elapsed"]
+        assert summary["avg"]
+        assert len(history) == 50
+        assert history[-1]["steps"] == 49
+    finally:
+        agent_process.kill()
+        run_cmd("rm -r artifacts")
+
+
+@pytest.mark.timeout(180)
+def test_kubernetes_agent_in_cluster(api_key, base_url):
+    # get user's wandb API key and AWS creds
+    if not api_key:
+        api_key = get_wandb_api_key(base_url)
+    aws_id = os.getenv("AWS_ACCESS_KEY_ID")
+    aws_secret = os.getenv("AWS_SECRET_ACCESS_KEY")
+    aws_token = os.getenv("AWS_SESSION_TOKEN")
+    create_wandb_and_aws_secrets(NAMESPACE, api_key, aws_id, aws_secret, aws_token)
```

Comment:
In an EKS cluster the instance role should make it so we don't require any of these env vars to be set

Reply:
This is running on a local cluster. Do we want to change the test to use an EKS cluster?

Reply:
We will want to add an EKS cluster set of tests, but we can maintain this as well, it provides value. We should however modify these secrets to be of the form that the launch system expects, which is from the `~/.aws/credentials` file